### ISSUE TITLE
@Transform not working

### ISSUE THREAD

**Issue by odejonge** [typestack/class-transformer#63]
With much pleasure I use class-transformer to decorate my model classes. Great job! Only when I try to use @Transform decorator in a model class it doesn't do what I need:

To meet the requirements of a REST service I need only to send the id of the nested object 'User' 

    export class Workorder
    {
        @Type(() => Number)
        public id:number
        ...
        @Type(() => User)
        @Transform(value => value.id, { toPlainOnly: true })
        mechanic:User
        ...
    }

needed for REST:

    {"workorder": 
          {
          ..., 
          "mechanic": 6537, 
          ...
          }
      }

I use the classToPlain function to create the asynchronous POST request

    let wo = classToPlain(workorder)
    let post = {workorder: wo}
    this.http.post(url, post)
        .map(res => res.json())
                .subscribe(json => {...})

Ionic compiles fine but I get this runtime error.

```
ERROR TypeError: Cannot read property 'prototype' of undefined
    at http://localhost/stedin/www/build/main.js:107310:84
    at Array.filter (native)
    at MetadataStorage.findMetadatas (http://localhost/stedin/www/build/main.js:107310:46)
    at MetadataStorage.findTransformMetadatas (http://localhost/stedin/www/build/main.js:107214:21)
    at TransformOperationExecutor.applyCustomTransformations (http://localhost/stedin/www/build/main.js:106904:58)
    at _loop_1 (http://localhost/stedin/www/build/main.js:106872:41)
    at TransformOperationExecutor.transform (http://localhost/stedin/www/build/main.js:106894:17)
    at _loop_1 (http://localhost/stedin/www/build/main.js:106871:45)
    at TransformOperationExecutor.transform (http://localhost/stedin/www/build/main.js:106894:17)
    at ClassTransformer.plainToClass (http://localhost/stedin/www/build/main.js:48087:25)
```
When I comment out the @Transform decorator the runtime error is gone.
Am I missing something here or is this an error in the package?

**Comment by gempain** [comment]
Same issue here with version 0.1.7:

```
 TypeError: Cannot read property 'prototype' of undefined
     at /Users/geoffroy/dev/git/ezvat/server/src/metadata/MetadataStorage.ts:161:69
     at Array.filter (native)
     at MetadataStorage.findMetadatas (/Users/geoffroy/dev/git/ezvat/server/src/metadata/MetadataStorage.ts:161:48)
     at MetadataStorage.findTransformMetadatas (/Users/geoffroy/dev/git/ezvat/server/src/metadata/MetadataStorage.ts:46:21)
     at TransformOperationExecutor.applyCustomTransformations (/Users/geoffroy/dev/git/ezvat/server/src/TransformOperationExecutor.ts:188:48)
     at _loop_1 (/Users/geoffroy/dev/git/ezvat/server/src/TransformOperationExecutor.ts:163:39)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at _loop_1 (/Users/geoffroy/dev/git/ezvat/server/src/TransformOperationExecutor.ts:162:43)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at _loop_1 (/Users/geoffroy/dev/git/ezvat/server/src/TransformOperationExecutor.ts:162:43)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at /Users/geoffroy/dev/git/ezvat/server/src/TransformOperationExecutor.ts:40:40
     at Array.forEach (native)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/src/TransformOperationExecutor.ts:37:30)
     at ClassTransformer.classToPlain (/Users/geoffroy/dev/git/ezvat/server/src/ClassTransformer.ts:19:25)
     at Object.classToPlain (/Users/geoffroy/dev/git/ezvat/server/src/index.ts:17:29)
     at ExpressDriver.handleSuccess (/Users/geoffroy/dev/git/ezvat/server/src/driver/express/ExpressDriver.ts:227:22)
```

**Comment by gempain** [comment]
Same here (version 0.1.7):

```
 TypeError: Cannot read property 'prototype' of undefined
     at metadata/MetadataStorage.ts:161:69
     at Array.filter (native)
     at MetadataStorage.findMetadatas (metadata/MetadataStorage.ts:161:48)
     at MetadataStorage.findTransformMetadatas (metadata/MetadataStorage.ts:46:21)
     at TransformOperationExecutor.applyCustomTransformations (TransformOperationExecutor.ts:188:48)
     at _loop_1 (TransformOperationExecutor.ts:163:39)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at _loop_1 (TransformOperationExecutor.ts:162:43)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at _loop_1 (TransformOperationExecutor.ts:162:43)
     at TransformOperationExecutor.transform (/Users/geoffroy/dev/git/ezvat/server/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at TransformOperationExecutor.ts:40:40
     at Array.forEach (native)
     at TransformOperationExecutor.transform (TransformOperationExecutor.ts:37:30)
     at ClassTransformer.classToPlain (ClassTransformer.ts:19:25)
     at Object.classToPlain (index.ts:17:29)
     at ExpressDriver.handleSuccess (driver/express/ExpressDriver.ts:227:22)
```

**Comment by gempain** [comment]
Same here with version 0.1.7 and Angular 4 (`angular-cli` based project):

```
 TypeError: Cannot read property 'prototype' of undefined
     at metadata/MetadataStorage.ts:161:69
     at Array.filter (native)
     at MetadataStorage.findMetadatas (metadata/MetadataStorage.ts:161:48)
     at MetadataStorage.findTransformMetadatas (metadata/MetadataStorage.ts:46:21)
     at TransformOperationExecutor.applyCustomTransformations (TransformOperationExecutor.ts:188:48)
     at _loop_1 (TransformOperationExecutor.ts:163:39)
     at TransformOperationExecutor.transform (/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at _loop_1 (TransformOperationExecutor.ts:162:43)
     at TransformOperationExecutor.transform (/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at _loop_1 (TransformOperationExecutor.ts:162:43)
     at TransformOperationExecutor.transform (/node_modules/class-transformer/TransformOperationExecutor.js:169:17)
     at TransformOperationExecutor.ts:40:40
     at Array.forEach (native)
     at TransformOperationExecutor.transform (TransformOperationExecutor.ts:37:30)
     at ClassTransformer.classToPlain (ClassTransformer.ts:19:25)
     at Object.classToPlain (index.ts:17:29)
     at ExpressDriver.handleSuccess (driver/express/ExpressDriver.ts:227:22)
```

**Comment by gempain** [comment]
I just found a solution. I was having this issue with a moment js transformer. The docs shows an example which does not work if not using `@Type(() => Date)` on the decorated property.

**Comment by Poitrin** [comment]
I tried @gempain's solution, but it still shows the same error message. Can anyone else confirm?

**Comment by gempain** [comment]
To be honest, after reading #28 I decided to remove my transformers. I see in the code where the problem is but haven't been able to find time to come up with a fix yet, though one was proposed in the issue I just mentioned.

**Comment by NoNameProvided** [comment]
I think this has been fixed in #100.

I will leave this issue open for now. After the next release please report it the issue still exists.
